### PR TITLE
:bug: [NPU] Fix Qwen3.5 dense serving on Ascend 950

### PR DESCRIPTION
**Pull request by TallMessiWu**
## Motivation

Two separate defects stop Qwen3.5 dense models from serving on Ascend 950. Each section below covers one; they are independent and can be reviewed separately.

This PR also carried a fix for the NPU graph rebind ordering in `NPUCudaGraphBackend.replay_with_input_update`, where the replay was issued while the input rebind was still in flight and `rtModelExecute` failed intermittently. #39589 has since fixed the same race upstream, and more completely -- it blocks on the update future before replaying and reuses one device-bound worker instead of creating a thread per replay -- so that fix is dropped here and the file now matches main.

Qwen3.5 uses Gemma RMSNorm. Ascend 950 has no registered kernel for `torch_npu.npu_gemma_rms_norm`, so the previous direct NPU path prevents Qwen3.5 from loading and serving on Ascend 950.

This PR is the SGLang half of the paired fix with sgl-project/sgl-kernel-npu#638. `sgl-kernel-npu` owns build-time SoC dispatch and exposes stable APIs; SGLang does not detect the SoC or choose providers.

## Modifications

- keep `KernelBackend.SGL_KERNEL_NPU` as the only Gemma RMSNorm backend on Ascend
- remove Gemma `TORCH_NPU` registration and the duplicated native forward path
- import `npu_gemma_rms_norm` from `norm.gemma_rmsnorm` for the ordinary path and preserve the existing `norm.add_rmsnorm_bias.add_gemma_rms_norm` residual API
- keep SoC/provider probing out of SGLang; preserve the legacy `srt` import fallback to `torch_npu.npu_gemma_rms_norm` for pre-provider 910 wheels, while an explicitly selected `SGL_KERNEL_NPU` backend does not silently fall back when its stable API is unavailable
- report an actionable package/version error if the stable API is unavailable
- preserve the established ordinary and residual call signatures in `GemmaRMSNorm` and `Gemma3RMSNorm`
- preserve the explicit pure-Torch debugging switch

## Paired architecture

```text
SGLang GemmaRMSNorm
  -> ordinary: sgl-kernel-npu npu_gemma_rms_norm
       -> 910 wheel (A2/A3): native Gemma ACLNN
       -> 950 wheel: standard RMSNorm ACLNN with 1 + weight
  -> residual: sgl-kernel-npu add_gemma_rms_norm
       -> all target wheels: add RMSNorm ACLNN with 1 + weight
```

For Ascend support, this PR requires a target-specific package built from sgl-project/sgl-kernel-npu#638; its Gemma-specific provider staging has since been generalized into generic target-provider staging by sgl-project/sgl-kernel-npu#734, with the stable operator API unchanged. Provider selection occurs only while building that wheel; `910` is the unified logical target for both A2 and A3.

## Additional fix: MXFP8 scale placeholder for partial blocks

`ModelSlimMXFP8Scheme.create_weights` sizes the weight-scale placeholder as `input_size_per_partition // 32`, which drops the trailing partial block whenever `K` is not a multiple of 32, and `NPUMXFP8LinearMethod.process_weights_after_loading` then pairs the scales with `reshape(n, k // 2, 2)`, which cannot consume an odd scale count.

Both are reachable on a real checkpoint. A ModelSlim `W8A8_MXFP8` export of Qwen3.5-27B stores the vision MLP down projection as:

```
model.visual.blocks.0.mlp.linear_fc2.weight        [1152, 4304]   W8A8_MXFP8
model.visual.blocks.0.mlp.linear_fc2.weight_scale  [1152,  135]   W8A8_MXFP8
```

`ceil(4304 / 32) = 135`, while the placeholder allocates 134, so the parameter fails to load; and 135 is odd, so the pair reshape fails as well. The placeholder now rounds up and an odd scale count is padded before the pair reshape. Layers whose `K` is a multiple of 32 are unaffected (the same block's `linear_fc1` has `K=1152` and 36 scales).

Reaching the vision tower at all additionally requires #36426, which forwards the ModelSlim config to the Qwen3-VL vision encoder; that PR is independent and not included here. The rounding is a correctness fix on its own: any MXFP8 linear with `K % 32 != 0` hits it.

## Accuracy Tests

- registry/mock coverage verifies that Ascend Gemma operations register only `SGL_KERNEL_NPU`, preserve the generic `out=` contract, and fail clearly on missing/incompatible packages
- delegation tests cover the established `npu_gemma_rms_norm(...)->(output, rstd)` and `add_gemma_rms_norm(...)->(norm_output, residual_sum)` contracts for `GemmaRMSNorm` and `Gemma3RMSNorm`
- `test/registered/unit/layers/quantization/test_modelslim_mxfp8.py`: the scale placeholder rounds up for `K=4304` and the post-load pair layout pads the odd 135th scale
- `py_compile`, Black, isort, and `git diff --check` passed on Windows
- Ascend 950PR_958b ordinary-path correctness: 13 cases passed
- Ascend 950PR_958b residual-add correctness: 32/32 FP16/BF16 cases passed; residual sums matched exactly and normalized outputs passed `atol=2e-2, rtol=2e-2`

## Speed Tests and Profiling

Both provider comparisons used Ascend 950PR_958b, 50 warmups, 200 synchronized p50 iterations, FP16/BF16, rows `1/16/128/512`, and hidden sizes `256/2048/4096/5120`.

- ordinary Triton geometric speedup versus `npu_rms_norm(input, 1 + weight)`: **-20.25%**; worst regression **82.34%**
- residual-add geometric `Triton / ACLNN` p50 latency ratio: **1.7765x**
- residual-add ACLNN p50 latency reduction relative to Triton: **43.7%**
- both Triton performance gates: **FAIL**

Therefore, sgl-kernel-npu #638 uses target-specific ACLNN providers for the ordinary Gemma RMSNorm path: native `npu_gemma_rms_norm(input, weight, eps)` on the 910 target (A2/A3), and standard `npu_rms_norm(input, 1 + weight, eps)` on the 950 target. The residual-add path uses `npu_add_rms_norm(..., 1 + weight, eps)` on all supported Ascend targets. SGLang contains no shape- or SoC-based runtime dispatch.

---
### CI States

Latest PR Test (Base): :x: [Run #35495115686](https://github.com/sgl-project/sglang/actions/runs/35495115686)
Latest PR Test (Extra): :x: [Run #35495115651](https://github.com/sgl-project/sglang/actions/runs/35495115651)
Latest PR Test (AMD ROCm 10): :x: [Run #35495115643](https://github.com/sgl-project/sglang/actions/runs/35495115643)

### CODE REVIEW

File: python/sglang/srt/utils/common.py [hunk 1]
```diff
@@ -185,6 +185,24 @@ def is_npu() -> bool:
...
+    """Whether the current NPU is an Ascend A5 (Ascend950) device.
+
+    Some fused ops (e.g. npu_gemma_rms_norm) have no kernel registered for the
+    A5 soc and must fall back to an equivalent path, while A2/A3 keep the op.
+    """
+    if not is_npu():
+        return False
+    import torch_npu  # noqa: F401
+
+    return torch.npu.get_soc_version() == _ASCEND_A5_SOC_VERSION
```
Comment (ping1jing2) [inline python/sglang/srt/utils/common.py]: we are not supposed to add such func here, maybe we should reuse `get_min_capability()`?

Comment (TallMessiWu) [reply python/sglang/srt/utils/common.py]: Removed.
